documentation/Plugins.md

```diff
@@ -0,0 +1,217 @@
+# Plugins
+
+There are two types of plugins.
+The ones that help building certain type systems with Manta Style, and the ones that help mocking data.
+
+## Mockers
+
+<!-- TODO: separate pages -->
+
+### `plugin-mock-example`
+
+Provides Manta Style examples for returns. Currently supports `string` and `number`.
+
+#### Installation
+
+```sh
+$ yarn add @manta-style/plugin-mock-example
+```
+
+#### Usage
+
+##### `string`
+
+Manta Style reads all `@example` strings and returns one of them at random.
+
+```js
+/**
+ * @example tanhauhau
+ * @example cryrivers
+ * @example jennieji
+ */
+userName: string; // returns 'tanhauhau', 'cryrivers', or 'jennieji' at random
+```
+
+#### `number`
+
+Manta Style returns a random value between all `@example` numbers rounded to `@precision` unit. If no `@precision` is provided, Manta Style will use 0, returning an integer. If Manta Style sees only one `@example` number, it will return that number.
+
+```js
+/**
+ * @example 5
+ * @example 10
+ * @precision 2
+ */
+score: number; // 6.27
+```
+
+### `plugin-mock-faker`
+
+Generate mock data using [faker.js](https://github.com/marak/faker.js/)
+
+#### @manta-style/plugin-mock-faker
+
+#### Installation
+
+```sh
+$ yarn add @manta-style/plugin-mock-faker
+```
+
+#### Usage
+
+For strings, our Mock Faker users Faker.js's `Faker.fake()`.
+
+Mock Faker is transparent to Faker.js's. You may use any available Faker.js's API.
+
+You can check out Faker.js's `Faker.fake()` API [here](http://marak.github.io/faker.js/#toc7__anchor)
+
+For numbers, `plugin-mock-faker` calls `faker.method.path()`.
```

Comment:
it calls the actual method in `faker`. for example, if you provide `date.past`, it calls `faker.date.past`.